src/features/user/Settings/EditProfile/AddressManagement/AddAddress.tsx

```diff
@@ -23,6 +22,11 @@ export type FormData = {
   state: string | null;
 };
 
+export type AddAddressApiPayload = FormData & {
+  country: ExtendedCountryCode | string;
+  type: 'other' | 'primary';
+};
```

Comment:
This type allows us to pass mandatory fields as null/undefined which is undesired. I suggest decoupling it from `FormData.`

Reply:
I have updated the `FormData `type to no longer accept `null or undefined `for the mandatory fields.